pkg/controller/operators/catalog/operator.go

```diff
@@ -1284,26 +1296,31 @@ func (o *Operator) setSubsCond(subs []*v1alpha1.Subscription, condType v1alpha1.
 		cond.Reason = reason
 		cond.Message = message
 		if setTrue {
+			if cond.Status == corev1.ConditionTrue {
+				continue
+			}
 			cond.Status = corev1.ConditionTrue
 		} else {
+			if cond.Status == corev1.ConditionFalse {
+				continue
+			}
 			cond.Status = corev1.ConditionFalse
 		}
 		sub.Status.SetCondition(cond)
 
```

Comment:
What do you think about splitting this function here: something like `setConditions(subs)` and `updateStatuses(subs)`

Then you can use `setConditions` over `updatedSubs` and `setConditions` + `updateStatus` over `allSubs` (right now this is hitting the api twice for all `updatedSubs`)

(I think there's bonus points if you find a nice way to align this with the existing `updateSubscriptionStatus` method)

Reply:
@ecordell I liked that idea. Please review the latest commit.